crates/bitwarden-core/src/platform/client_platform.rs

```diff
@@ -5,15 +5,54 @@ use super::{
 };
 use crate::{error::Result, Client};
 
+/// A struct containing platform utilities.
 pub struct ClientPlatform<'a> {
     pub(crate) client: &'a Client,
 }
 
 impl<'a> ClientPlatform<'a> {
+    /// Will generate a fingerprint based on the `input`. Given the same `input` This
+    /// method will result in the exact same output.
+    ///
+    /// # Examples
+    /// ```rust
+    /// use bitwarden_core::{Client, platform::FingerprintRequest};
+    ///
+    /// async fn test() {
+    ///     let client = Client::test_account().await;
+    ///     let fingerprint_response = client.platform()
+    ///         .fingerprint(&FingerprintRequest {
+    ///             fingerprint_material: "my_material".to_owned(),
+    ///             public_key: "...public key...".to_owned(),
+    ///         })
+    ///         .unwrap();
+    ///
+    ///     println!("{}", fingerprint_response.fingerprint);
+    /// }
+    /// ```
     pub fn fingerprint(&self, input: &FingerprintRequest) -> Result<FingerprintResponse> {
         generate_fingerprint(input)
     }
 
+    /// Will generate a fingerprint based on the given `fingerprint_material`
+    /// and the users public key. Given the same `fingerprint_material` and
+    /// the same user. This method will result in the exact same output.
+    ///
+    /// The returned fingerprint is a string of 5 words seperated by hyphens.
+    ///
+    /// # Examples
+    /// ```rust
+    /// use bitwarden_core::Client;
+    ///
+    /// async fn test() {
+    ///     let client = Client::test_account().await;
+    ///     let fingerprint = client.platform()
+    ///         .user_fingerprint("my_material".to_owned())
+    ///         .unwrap();
+    ///
+    ///     assert_eq!(fingerprint, "dreamland-desecrate-corrosive-ecard-retry");
```

Comment:
thought: huh, interesting, here we use `assert_eq` to give an example of the output, but in the doc above we use `println`. Should we choose one approach for consistency maybe? I like this one with `assert_eq`

Reply:
Ideally we'd just assert! as much as possible, that way we can have an explanatory code comment and test all in one. 

That might not be always acceptable, if the output of the function is too long we don't want the code sample to be dwarfed by a massive multi-line assert for example.

Reply:
Yeah I also prefer assert but the one above didn't use it in the above doc because I couldn't come up with a way to concisely show off a public key that wouldn't be pretty ugly for a test. I could probably generate a new one in a simple one-liner but I still couldn't `assert_eq!` it because it would be different each time and `assert_ne(fingerprint, "")` doesn't seem like something we want to do.